src/models/blocks/kit.rs

```diff
@@ -29,6 +29,8 @@ pub enum SlackBlock {
     Input(SlackInputBlock),
     #[serde(rename = "file")]
     File(SlackFileBlock),
+    #[serde(rename = "event")]
+    Event(SlackFileBlock),
```

Comment:
Why we have `SlackFileBlock` type here. Is this correct?

Reply:
This was something that was missing when I was going through all the data I have in my Slack workspace. I am still going through every message in my slack workspace ~500k and I found some more things that needed to be added to SlackMessageEvent enum. I can remove this SlackFileBlock and add it to another issue if you'd like. And also add more SlackMessageEvent enums to this issue

Reply:
This doesn't look correct. You're trying to introduce a new block called event, but it doesn't exist in documentation:
https://api.slack.com/reference/block-kit/blocks. Can you show me the example and docs reference where you seen this?

Reply:
I found the JSON of the message that is causing this though it is not matching up to the error I am seeing for SlackClientProtocolError so maybe this is inconsistent and I am wrong but there is still an issue in here somewhere.

Here is the Error that is getting printed when trying to use this conversations.history method

```Error: ProtocolError(SlackClientProtocolError { json_error: Error("unknown variant `event`, expected one of `section`, `header`, `divider`, `image`, `actions`, `context`, `input`, `file`, `rich_text`"```

Here is the JSON for the message that is causing this

```
{
    "ok": true,
    "latest": "1664387316.198939",
    "messages": [
        {
            "client_msg_id": "320DC39E-5924-4797-BB63-3CFED83C2AC4",
            "type": "message",
            "text": "That’s just convenience for me.",
            "user": "WFH0F0873",
            "ts": "1664387315.198939",
            "blocks": [
                {
                    "type": "rich_text",
                    "block_id": "ztW6",
                    "elements": [
                        {
                            "type": "rich_text_section",
                            "elements": [
                                {
                                    "type": "text",
                                    "text": "That’s just convenience for me."
                                }
                            ]
                        }
                    ]
                }
            ],
            "team": "T2N76FZ3Q"
        }
    ],
    "has_more": true,
    "pin_count": 12,
    "channel_actions_ts": null,
    "channel_actions_count": 0,
    "response_metadata": {
        "next_cursor": "bmV4dF90czoxNjY0Mzg3MzA3NTIxOTg5"
    }
}
        ```

Although there isn't an event key in that JSON block

Reply:
That's interesting. It feels like the error is related to some of the JSON we can't actually see.
It is possible they have introduced one more undocumented block, can you put this instead of `SlackFileBlock`

````
#[serde(rename = "event")]
 Event(serde_json::Value),
````
and try to get some example of it, please?

Reply:
That's interesting. It feels like the error is related to some of the JSON we can't actually see.
It is possible they have introduced one more undocumented block, can you put this instead of `SlackFileBlock`

````
#[serde(rename = "event")]
 Event(serde_json::Value),
````
and try to get some example of it, please?

Reply:
It doesn't throw the error now that is sees the event type there. It returns this JSON form of the SlackHistoryMessage
```
Problem message: SlackHistoryMessage { origin: SlackMessageOrigin { ts: SlackTs("1664387315.198939"), channel: None, channel_type: None, thread_ts: None, client_msg_id: Some(SlackClientMessageId("320DC39E-5924-4797-BB63-3CFED83C2AC4")) }, content: SlackMessageContent { text: Some("That’s just convenience for me."), blocks: Some([RichText(Object {"block_id": String("ztW6"), "elements": Array [Object {"elements": Array [Object {"text": String("That’s just convenience for me."), "type": String("text")}], "type": String("rich_text_section")}]})]), attachments: None, upload: None, files: None, reactions: None }, sender: SlackMessageSender { user: Some(SlackUserId("WFH0F0873")), bot_id: None, username: None, display_as_bot: None }, parent: SlackParentMessageParams { reply_count: None, reply_users_count: None, latest_reply: None, reply_users: None }, subtype: None }
```

Reply:
Are you sure this is the problematic message? Since it doesn't contain any `event` in it either. This is off. We are supposed to see some message with a block in it with the type `event`.

Reply:
Apparently it was a giant stack trace. I found the actual JSON

```
SlackHistoryMessage { origin: SlackMessageOrigin { ts: SlackTs("1664317158.245199"), channel: None, channel_type: None, thread_ts: None, client_msg_id: None }, content: SlackMessageContent { text: Some("This content can't be displayed."), blocks: Some([Event(Object {"block_id": String("1"), "event_id": String("aaaabaynv6otdov6gocrppc5r4:of5kjuvgvhqhdkjnlvb7zxp63u"), "is_deleted": Bool(true)})]), attachments: None, upload: None, files: None, reactions: None }, sender: SlackMessageSender { user: Some(SlackUserId("WGJRL9NJD")), bot_id: None, username: None, display_as_bot: None }, parent: SlackParentMessageParams { reply_count: None, reply_users_count: None, latest_reply: None, reply_users: None }, subtype: None }
```
Interestingly enough, this came from my account and I am not sure what I did to this message to make this happen. It just says `We can't fully display this message.`

Reply:
It looks like I deleted it but it is still there

Reply:
> It looks like I deleted it but it is still there

It looks like I deleted it in the JSON, but it is still there in the slack chat and I have the option to delete it in the UI

Reply:
Alright. Then this is one more undocument block type.
Let's have it defined as:
```
#[serde(rename = "event")]
 Event(serde_json::Value),
```
Because we can't define its structure without reliable source or docs for now. And I'll merge your PR afterwards 

Thanks for digging with me and the contributions 👍🏻

Reply:
Ok I have done this and left it under the undocumented block type comment